code/controllers/subsystem/processing/greyscale.dm

```diff
@@ -27,12 +27,66 @@ PROCESSING_SUBSYSTEM_DEF(greyscale)
 		var/datum/greyscale_config/config = configurations[greyscale_type]
 		config.CrossVerify()
 
+	if(CONFIG_GET(flag/generate_assets_in_init))
+		ExportMapPreviews()
+
 	return SS_INIT_SUCCESS
 
 /datum/controller/subsystem/processing/greyscale/proc/RefreshConfigsFromFile()
 	for(var/i in configurations)
 		configurations[i].Refresh(TRUE)
 
+/datum/controller/subsystem/processing/greyscale/proc/ExportMapPreviews()
+	// Put subtypes before their parent or the parent file will take all the generated icons
+	var/static/list/types_that_get_their_own_file = list(
+		"turfs" = /turf, // None of these yet but it's harmless to be prepared
+		"mobs" = /mob, // Ditto
+		"clothing" = /obj/item/clothing,
+		"items" = /obj/item,
+		"objects" = /obj,
```

Comment:
am I insane or are we doing duped work here

Reply:
types are skipped if they've been handled already